grammar.js

```diff
@@ -922,12 +923,22 @@ module.exports = grammar({
       $.negated_pattern,
       $.parenthesized_pattern,
       $.relational_pattern,
-      $.binary_pattern,
+      $.or_pattern,
+      $.and_pattern,
+      $.list_pattern,
       $.type_pattern
     ),
 
     type_pattern: $ => $._type,
 
+    list_pattern: $ => seq(
+      '[',
+      optional(seq(commaSep(choice($._pattern, $.slice_pattern)), optional(','))),
```

Comment:
Should this be `commaSep1`? Otherwise, a single comma without preceding elements is allowed, I think.

Reply:
Good catch, I must have thought `[,]` was legal syntax but it is not so I can switch that out.

Reply:
Fixed in latest commit.